include/base_interface.h

```diff
@@ -0,0 +1,322 @@
+   * It takes as argument the number of DoF per cell @p dofs_per_cell,
+   * the number of quadrature points @p n_q_points, the number of
+   * quadrature points per face @p n_face_q_points, the reference to
+   * solutions vectors @p sol and the reference to the AnyData @p d.
+   *
+   * TODO: add current_time and current_alpha
+   */
+  virtual void initialize_data(const typename LAC::VectorType &solution,
+                               const typename LAC::VectorType &solution_dot,
+                               const double t,
+                               const double alpha) const;
+
+
+  /**
+   * Definition of energies and residuals
+   */
+  virtual void get_energies_and_residuals(const typename DoFHandler<dim,spacedim>::active_cell_iterator &,
+                                          FEValuesCache<dim,spacedim> &,
+                                          std::vector<Sdouble> &energies,
+                                          std::vector<std::vector<double> > &local_residuals,
+                                          bool compute_only_system_matrix) const;
+
+  /**
+   * Definition of energies and residuals
+   */
+  virtual void get_energies_and_residuals(const typename DoFHandler<dim,spacedim>::active_cell_iterator &,
+                                          FEValuesCache<dim,spacedim> &,
+                                          std::vector<SSdouble> &energies,
+                                          std::vector<std::vector<Sdouble> > &local_residuals,
+                                          bool compute_only_system_matrix) const;
+
+  /**
+   * This function can be overloaded to directly implement the local
+   * matrices (i.e. as it is usally done in standard FE codes)
+   */
+  virtual void assemble_local_matrices (const typename DoFHandler<dim,spacedim>::active_cell_iterator &cell,
+                                        FEValuesCache<dim,spacedim> &scratch,
+                                        CopyData &data) const;
+
+  virtual void get_local_system_residual (const typename DoFHandler<dim,spacedim>::active_cell_iterator &cell,
+                                          FEValuesCache<dim,spacedim> &scratch,
+                                          CopyData &data) const;
+
+  /**
+   * Compute linear operators needed by the problem
+   *
+   * This function is used to assemble linear operators related
+   * to the problem.
+   */
+  virtual void compute_system_operators(const DoFHandler<dim,spacedim> &,
+                                        const std::vector<shared_ptr<typename LAC::BlockMatrix> >,
+                                        LinearOperator<typename LAC::VectorType> &,
+                                        LinearOperator<typename LAC::VectorType> &) const;
+
+
+  shared_ptr<Mapping<dim,spacedim> > get_mapping() const;
+
+  virtual unsigned int set_mapping_degree() const;
+
+  virtual UpdateFlags get_face_update_flags() const;
+
+  virtual UpdateFlags get_matrices_update_flags() const;
```

Comment:
This is now called `get_matrices_update_flags`, while the above is called `get_face_update_flags`. I think the correct name here should be `get_cell_update_flags(unsigned int index=0)` and `get_face_update_flags(unsigned int index=0)`